src/vm/perfmap.cpp

```diff
@@ -218,6 +222,38 @@ void PerfMap::LogJITCompiledMethod(MethodDesc * pMethod, PCODE pCode, size_t cod
     }
 }
 
+// Log a set of stub to the map.
+void PerfMap::LogStubs(const char* stubType, const char* stubOwner, PCODE pCode, size_t codeSize)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    if (s_Current == nullptr)
+    {
+        return;
+    }
+
+    // Logging failures should not cause any exceptions to flow upstream.
+    EX_TRY
+    {
+        if(!stubOwner)
+        {
+            stubOwner = "?";
+        }
+        if(!stubType)
+        {
+            stubOwner = "?";
+        }
+
+        // Build the map file line.
+        SString line;
+        line.Printf("%p %x stub<%d> %s<%s>\n", pCode, codeSize, ++(s_Current->m_StubsMapped), stubType, stubOwner);
```

Comment:
@brianrob 
Recently they made coreclr use the os random address map feature.  This make is difficult to use the address as an identifier.  

I added the counter to make it fairly easy to set a repeatable breakpoint.  

I also added it to make the name guaranteed to be unique.  As I was going through these, it was rarely obvious on behalf of whom the work was being done at the point in the cal stack where the code was being generated.  The stub number helps locate the stub in the perf-*.map file.  The stub is typically related to the function before of after it.  The stubs are not allocated in address order since there are multiple code pools.  So address does not serve the same purpose, but file line number does.

I was doing this prototype style, so it got quick and dirty.  I am happy to revise....